packages/features/ee/managed-event-types/lib/handleChildrenEventTypes.ts

```diff
@@ -244,22 +242,39 @@ export default async function handleChildrenEventTypes({
       .reduce((newObj, [key, value]) => ({ ...newObj, [key]: value }), {});
     console.log({ unlockedFieldProps });
     // Update event types for old users
-    const oldEventTypes = await prisma.$transaction(
-      oldUserIds.map((userId) => {
-        return prisma.eventType.update({
-          where: {
-            userId_parentId: {
+    const oldEventTypes = await prisma.$transaction(async (prisma) => {
+      return await Promise.all(
+        oldUserIds.map(async (userId) => {
+          const existingEventType = await prisma.eventType.findFirst({
+            where: {
               userId,
               parentId,
             },
-          },
-          data: {
-            ...updatePayloadFiltered,
-            hashedLink: "hashedLink" in unlockedFieldProps ? undefined : hashedLinkQuery(userId),
-          },
-        });
-      })
-    );
+          });
+
+          const existingHashedLink = await prisma.hashedLink.findFirst({
+            where: {
+              eventTypeId: existingEventType ? existingEventType.id : undefined,
+            },
+          });
+
+          const eventType = await prisma.eventType.update({
+            where: {
+              id: existingEventType ? existingEventType.id : undefined,
+            },
```

Comment:
Also I don't follow what's the point of calling these in case exisitngEventType is undefined?

Reply:
@Amit91848 Yes correct! My bad.

I have added a condition now to run this if eventType is not `undefined`.